### PR TITLE
Make PopArt Work

### PR DESCRIPTION
**Pull request by yycho0108**
## Description
Make PopArt work.
There were a number of bugs in the previous implementation that made it disfunctional.
Now, the training seems to work-ish.

## Demo

Works on CartPole:
```bash
$ python3 test_pop_art_cartpole.py
$ # tensorboard --logdir /tmp/cartpole [--host 0.0.0.0]
```
Screenshot from 2022-06-12 13-25-06

## Changelog
* fix interoperability in collation
* automatically infer `num_envs` arg
* add weight decay to adam optimizer
* enable slightly off-policy RL with replay buffer
* add checkpoint save/load capability
* fix time dimension add/removal scheme for LSTM interface
* track cumulative episode rewards
* fix collation for reward, termination signals
* fix log-probability calculation for incorrect action samples
* add vtrace testing script adopted from deepmind repo
* enable popart update parameters
* add gradient histogram logging capability
* add cartpole testing script and make it work
* add a few utility scripts in `util.py`

### CODE REVIEW

**Review by Minchan-Kim — APPROVED** [review]
I approve this. Glad that it works on Cartpole.
